File: articles/overconfidence-bias-financial-markets/go/overconfidence_demand.go
package main

import "fmt"

func investorDemand(expectedReturn, perceivedVariance, alpha, beta float64) float64 {
	return alpha*expectedReturn - beta*perceivedVariance
}

func netReturnAfterCost(grossReturn, tradingIntensity, costPerTurnover float64) float64 {
	return grossReturn - costPerTurnover*tradingIntensity
}

func main() {
	fmt.Printf("Synthetic investor demand: %.3f\n", investorDemand(0.08, 0.03, 1.2, 0.7))
	fmt.Printf("Synthetic net return after cost: %.3f\n", netReturnAfterCost(0.05, 1.4, 0.0025))
}

File: articles/overconfidence-bias-financial-markets/cpp/overconfidence_demand.cpp
#include <iostream>

double investor_demand(double expected_return, double perceived_variance, double alpha, double beta) {
    return alpha * expected_return - beta * perceived_variance;
}

double net_return_after_cost(double gross_return, double trading_intensity, double cost_per_turnover) {
    return gross_return - cost_per_turnover * trading_intensity;
}

int main() {
    std::cout << "Synthetic investor demand: " << investor_demand(0.08, 0.03, 1.2, 0.7) << std::endl;
    std::cout << "Synthetic net return after cost: " << net_return_after_cost(0.05, 1.4, 0.0025) << std::endl;
    return 0;
}
